gravityflow/gravityflow
# Approval Step Revert Notification + User Input Assignee Notification Override

**This feature PR:**

1. Enhances the Approval Step Notifications to include a new tab for Revert message.
It is only visible when the Revert setting has been selected.
2. When Revert notification is active, by default the user input assignee notification is disabled.
3. Adds the gravityflow_user_input_notification_send_on_revert filter to allow user input notification to continue to be sent when the revert notification has been defined / sent.
4. Adds acceptance test (0028) which validates approval, revert and notifications sent

**To test this feature:**
- Execute automated test form/flow 0028 or follow manual step instructions below.
- Create a form which has an approval step followed by user input step.
- Activate the 'Revert to User Input step' option.
- See that the Revert Email tab in notifications is available.
- Define a Revert Email and save settings.
- Submit the form and revert the approval step during workflow.

If you add the following code snippet to functions.php and repeat the test procedure, both the user input assignee notification and the revert notification will be sent.

```
add_action(` 'gravityflow_user_input_notification_send_on_revert', 'sh_revert_override', 10, 5);
  function sh_revert_override( $override, $notification, $form, $entry, $step ) 
    return true;
  }
```
